services/provider/server/server.go

```diff
@@ -367,6 +370,38 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		}),
 	})
 
+	if consumerResource.Spec.StorageQuotaInGiB > 0 {
+		hardLimit := resource.MustParse(fmt.Sprintf("%d%s", consumerResource.Spec.StorageQuotaInGiB, "Gi"))
+		storageQuota := &quotav1.ClusterResourceQuota{
+			ObjectMeta: metav1.ObjectMeta{Name: "ocs-storagecluster-quota"},
+			Spec: quotav1.ClusterResourceQuotaSpec{
+				Selector: quotav1.ClusterResourceQuotaSelector{
+					LabelSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "volume.kubernetes.io/storage-provisioner",
+								Values:   []string{"openshift-storage.rbd.csi.ceph.com"},
+								Operator: metav1.LabelSelectorOpExists,
+							},
+						},
+					},
+				},
+				Quota: corev1.ResourceQuotaSpec{
+					Hard: corev1.ResourceList{"requests.storage": hardLimit},
+				},
+			},
+		}
+
+		extR = append(extR, &pb.ExternalResource{
+			Name: "QuotaForConsumer",
+			Kind: "ClusterResourceQuota",
+			Data: mustMarshal(map[string]string{
+				"spec": fmt.Sprintf("%+v\n", storageQuota),
```

Comment:
instead of sending the entire spec for ClusterResourceQuota we can send only a value for `StorageQuotaInGiB` and construct the ClusterResourceQuota on the client side. I am sending this as this is the initial PR and need your inputs

Reply:
Sending the entire resource so that the client does not need to create or set anything, just create it.